tests/functionaltests/Tests/CortanaTests.mm

```diff
@@ -106,9 +221,75 @@
         *kind = ActivationKind_VoiceCommand;
         return S_OK;
     });
-    
+
     fakeVoiceCommandActivatedEventArgs->Setget_PreviousExecutionState([](ApplicationExecutionState* state) {
         *state = ApplicationExecutionState_NotRunning;
         return S_OK;
     });
+
+    // Swizzle method to use our delegate for testing
+    Method originalMethod = class_getInstanceMethod([UIApplication class], @selector(setDelegate:));
+    Method swizzledMethod = class_getInstanceMethod([UIApplication class], @selector(swizzleVoiceCommandDelegate:));
+    method_exchangeImplementations(originalMethod, swizzledMethod);
```

Comment:
Though this is a cool way to test, all that was required was to set g_delegateClassName with the classname of your custom delegate in StarboardXaml.cpp for it to be picked up. This is how UIApplicationDelegate gets set and would avoid test restriction to be able to call it only once per class etc.

Your call if you want to clean it up and make the test simpler.